src/map/battlefield.cpp

```diff
@@ -330,7 +336,12 @@ bool CBattlefield::InsertEntity(CBaseEntity* PEntity, bool enter, BATTLEFIELDMOB
 
             if (pet && pet->PMaster && pet->PMaster->objtype == TYPE_PC)
             {
-                // dont enter player pet
+                // Only allow Wyvern pets to enter
+                if (pet->getPetType() == PET_TYPE::WYVERN)
```

Comment:
Based on retail observations, it seems _all_ pets are allowed, even if we erroneously despawn some of them on level sync right now, but that's out of scope for this PR.

Reply:
I was just writing out a comment stating exactly this. I will remove the check for wyverns and we can deal with not despawning the pets during level sync in another PR in the future. Is this something that SE changed at some point?

Reply:
I only played DRG throughout retail, so I can only comment that it always worked for DRG based on memory. Can't say for anything else.